corehq/apps/locations/forms.py

```diff
@@ -650,21 +663,36 @@ def __init__(self, domain, location, *args, **kwargs):
         self.helper.label_class = 'col-sm-3 col-md-2'
         self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
 
+    def clean(self):
+        cleaned_data = super(RelatedLocationForm, self).clean()
+        for name, value in cleaned_data.items():
+            if name.startswith('relation_distance_') and value and value < 0:
+                raise forms.ValidationError("The distance cannot be a negative value")
```

Comment:
Does the javascript assure that `value` is an int?

Reply:
I don't think it uses any javascript https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number